Source/BoundaryConditions/WarpX_PEC.cpp

```diff
@@ -62,9 +68,13 @@ PEC::ApplyPECtoEfield (std::array<std::unique_ptr<amrex::MultiFab>, 3>& Efield,
         amrex::Array4<amrex::Real> const& Ez = Efield[2]->array(mfi);
 
         // Extract tileboxes for which to loop
-        amrex::Box const& tex = mfi.tilebox(Efield[0]->ixType().toIntVect(), shape_factor);
-        amrex::Box const& tey = mfi.tilebox(Efield[1]->ixType().toIntVect(), shape_factor);
-        amrex::Box const& tez = mfi.tilebox(Efield[2]->ixType().toIntVect(), shape_factor);
+        // if split field, the box includes nodal flag, else nodal flag plus particle shape factor
+        amrex::Box const& tex = (split_pml_field) ? mfi.tilebox(Efield[0]->ixType().toIntVect())
+                                                  : mfi.tilebox(Efield[0]->ixType().toIntVect(), shape_factor);
+        amrex::Box const& tey = (split_pml_field) ? mfi.tilebox(Efield[1]->ixType().toIntVect())
+                                                  : mfi.tilebox(Efield[1]->ixType().toIntVect(), shape_factor);
+        amrex::Box const& tez = (split_pml_field) ? mfi.tilebox(Efield[2]->ixType().toIntVect())
+                                                  : mfi.tilebox(Efield[2]->ixType().toIntVect(), shape_factor);
```

Comment:
Could you remind of what is the effect of passing the additional argument `shape_factor` to the `tilebox` method here?

Reply:
for the Efields used in the valid region, when we pass the shape factor, it will "grow" the box into the guard cell region by the number of indices equal to the shape factor. We need to include this in addition to just the nodality because particles gather fields from these cells as well. And the guard cells are treated separately in the PEC module.

Reply:
Thank you for the clarification, Reva! I wonder then if it would be safer to grow this by a number of guard cells equal to the `IntVect` `ng_FieldGather` that we have in Source/Parallelization/GuardCellManager.H (which would probably require to defined some "getter" similar to `getngE()` and others that already exist in the `WarpX` class). What do you think? Would probably be worth looking briefly at how `ng_FieldGather` is computed in the `guardCellManager` class, it is probably equal to the shape factor by default, but might be taking into account some other algorithmic variants as well.

Reply:
Thank you Edoardo. we use ng_FieldGather now. :)